src/Providers/Stripe/Payment/Service.php

```diff
@@ -88,14 +97,56 @@ public function create(
                 $response = PaymentIntent::retrieve($token->value);
                 $response->confirm();
 
+                break;
+            case StripeToken::PAYMENT_METHOD:
+                if ($customer && $customer->type !== StripeToken::CUSTOMER) {
+                    throw new \Exception('You need to pass a valid customer token to charge a card one.');
+                }
+
+                /** @var PaymentIntent $response */
+                $response = PaymentIntent::create([
+                    'payment_method' => $token->value,
+                    'customer' => $customer->value,
+                    'amount' => $amount,
+                    'currency' => $currency,
+                    'confirmation_method' => 'manual',
+                    'confirm' => true,
+                    "payment_method_types" => ["card"],
+                    'description' => $description,
+                    'metadata' => $metadata,
+                    'use_stripe_sdk' => true,
+                ]);
+
                 break;
         }
 
         if ($response instanceof PaymentIntent && $response->status !== PaymentIntent::STATUS_SUCCEEDED) {
-            throw new StripeException($response);
+            $token = new StripeToken($response->client_secret);
+
+            throw new StripeException($token);
         }
 
-        return new Payment($response->id, $response->currency, $response->amount, $response->created);
+        return new Payment($response->id, $response->amount, $response->currency, $response->created);
+    }
+
+    /**
+     * Return the name of the provider.
+     *
+     * @return string
+     */
+    public function getProviderName(): string
+    {
+        return 'stripe';
+    }
```

Comment:
should ditch this for sure

Reply:
Same as I explained earlier having this method prevent us from having 'stripe' statement within the PaymentService of the API3 when recording billing_events (`transaction_provider` field).